### PR TITLE
added required controllers,

### PR DESCRIPTION
**Pull request by AndriiBardin**
mappers, dto

### CODE REVIEW

**Review by yurwing — APPROVED** [review]

File: src/main/java/cinema/controller/CinemaHallController.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import java.util.List;
+import java.util.stream.Collectors;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/halls")
```
Comment (KseniiaMakarova) [inline src/main/java/cinema/controller/CinemaHallController.java]: please see task description - what should the mapping be here?

File: src/main/java/cinema/controller/CinemaHallController.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        CinemaHall cinemaHall = cinemaHallMapper.requestDtoToCinemaHAll(cinemaHallRequestDto);
+        cinemaHallService.add(cinemaHall);
+    }
+
+    @GetMapping
+    public List<CinemaHallResponseDto> getAll() {
+        return cinemaHallService.getAll().stream()
+                .map(cinemaHallMapper::cinemaHallToResponseDto)
+                .collect(Collectors.toList());
+
```
Comment (KseniiaMakarova) [inline src/main/java/cinema/controller/CinemaHallController.java]: redundant empty line

File: src/main/java/cinema/controller/MovieController.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        movieService.add(movie);
+    }
+
+    @GetMapping
+    public List<MovieResponseDto> getAll() {
+        return movieService.getAll().stream()
+                .map(movieMapper::movieToResponseDto)
+                .collect(Collectors.toList());
+    }
+
```
Comment (KseniiaMakarova) [inline src/main/java/cinema/controller/MovieController.java]: same

File: src/main/java/cinema/controller/MovieSessionController.java [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/sessions")
```
Comment (KseniiaMakarova) [inline src/main/java/cinema/controller/MovieSessionController.java]: same comment about mapping

File: src/main/java/cinema/dao/CinemaHallDao.java [hunk 5]
```diff
@@ -2,9 +2,12 @@
...
 import cinema.model.CinemaHall;
 import java.util.List;
+import java.util.Optional;
 
 public interface CinemaHallDao {
     CinemaHall add(CinemaHall cinemaHall);
 
     List<CinemaHall> getAll();
+
+    Optional<CinemaHall> getId(Long id);
```
Comment (KseniiaMakarova) [inline src/main/java/cinema/dao/CinemaHallDao.java]:
```suggestion
    Optional<CinemaHall> get(Long id);
```

File: src/main/java/cinema/dao/impl/CinemaHallDaoImpl.java [hunk 6]
```diff
@@ -50,4 +51,13 @@ public CinemaHall add(CinemaHall cinemaHall) {
             throw new DataBaseException("Can't get all movies", e);
         }
     }
+
+    @Override
+    public Optional<CinemaHall> getId(Long id) {
+        try (Session session = sessionFactory.openSession()) {
+            return Optional.ofNullable(session.get(CinemaHall.class, id));
+        } catch (Exception e) {
+            throw new DataBaseException("Can't get movie by id " + id, e);
```
Comment (KseniiaMakarova) [inline src/main/java/cinema/dao/impl/CinemaHallDaoImpl.java]: let's make the message informative

File: src/main/java/cinema/service/CinemaHallService.java [hunk 7]
```diff
@@ -7,4 +7,6 @@
     CinemaHall add(CinemaHall cinemaHall);
 
     List<CinemaHall> getAll();
+
+    CinemaHall getId(Long id);
```
Comment (KseniiaMakarova) [inline src/main/java/cinema/service/CinemaHallService.java]:
```suggestion
    CinemaHall get(Long id);
```

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/cinema/dao/impl/CinemaHallDaoImpl.java [hunk 8]
```diff
@@ -53,11 +53,11 @@ public CinemaHall add(CinemaHall cinemaHall) {
     }
 
     @Override
-    public Optional<CinemaHall> getId(Long id) {
+    public Optional<CinemaHall> get(Long id) {
         try (Session session = sessionFactory.openSession()) {
             return Optional.ofNullable(session.get(CinemaHall.class, id));
         } catch (Exception e) {
-            throw new DataBaseException("Can't get movie by id " + id, e);
+            throw new DataBaseException("Movie by id " + id + " does not exist", e);
```
Comment (KseniiaMakarova) [inline src/main/java/cinema/dao/impl/CinemaHallDaoImpl.java:60]: do you mean CinemaHall?

**Review by KseniiaMakarova — APPROVED** [review]

File: src/main/java/cinema/dao/impl/CinemaHallDaoImpl.java (hunk 8 above)
Comment (AndriiBardin) [reply src/main/java/cinema/dao/impl/CinemaHallDaoImpl.java:60]: fixed )

**Pull request merged**